File: boat-sample/src/test/java/sample/java/xyz/srclab/common/cache/CacheSample.java
package sample.java.xyz.srclab.common.cache;

import org.testng.annotations.Test;
import xyz.srclab.common.cache.Cache;
import xyz.srclab.common.test.TestLogger;

public class CacheSample {

    private static final TestLogger logger = TestLogger.DEFAULT;

    @Test
    public void testCache() {
        Cache<String, String> cache = Cache.newFastCache();
        cache.getOrLoad("1", k -> k);
        //1
        logger.log("1: {}", cache.get("1"));
        //null
        logger.log("2: {}", cache.getOrNull("2"));
    }
}

File: boat-sample/src/test/kotlin/sample/xyz/srclab/common/cache/CacheSample.kt
package sample.xyz.srclab.common.cache

import org.testng.annotations.Test
import xyz.srclab.common.cache.Cache
import xyz.srclab.common.test.TestLogger

class CacheSampleKt {

    @Test
    fun testCache() {
        val cache = Cache.newFastCache<String, String>()
        cache.getOrLoad("1") { k: String -> k }
        //1
        logger.log("1: {}", cache.get("1"))
        //null
        logger.log("2: {}", cache.getOrNull("2"))
    }

    companion object {
        private val logger = TestLogger.DEFAULT
    }
}